### ISSUE TITLE
Cannot select atoms with StringSelection from empty AtomGroup

### ISSUE THREAD

**Issue by lilyminium** [MDAnalysis/mdanalysis#2765]
## Expected behavior ##

`empty_atomgroup.select_atoms('name anything')` returns an empty AtomGroup.

## Actual behavior ##

A TypeError is raised.

## Code to reproduce the behavior ##

```python
In [1]: import MDAnalysis as mda

In [2]: from MDAnalysis.tests.datafiles import PSF

In [3]: mda.__version__
Out[3]: '2.0.0-dev0'

In [4]: u = mda.Universe(PSF)

In [5]: ag = u.select_atoms('resname blahblah')

In [6]: len(ag)
Out[6]: 0

In [7]: ag.select_atoms('name O')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-c43a12bdaf29> in <module>
----> 1 ag.select_atoms('name O')

~/pydev/mdanalysis/package/MDAnalysis/core/groups.py in select_atoms(self, sel, *othersel, **selgroups)
   2861             # Apply the first selection and sum to it
   2862             atomgrp = sum([sel.apply(self) for sel in selections[1:]],
-> 2863                           selections[0].apply(self))
   2864         return atomgrp
   2865

~/pydev/mdanalysis/package/MDAnalysis/core/selection.py in apply(self, group)
    517         for val in self.values:
    518             values = getattr(group, self.field)
--> 519             mask |= [fnmatch.fnmatch(x, val) for x in values]
    520         return group[mask].unique
    521

TypeError: ufunc 'bitwise_or' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev0
- Which version of Python (`python -V`)? 3.7
- Which operating system? MacOS

**Comment by RMeli** [comment]
I agree that the `TypeError` is really obscure, but wouldn't be better to have the selection on an empty atom group to fail cleanly instead of returning an empty atom group?

**Comment by richardjgowers** [comment]
I think it’s fairly common to chain selections, so an empty AG should
return another empty AG. I can’t remember, is there a way to go to universe
again via global or something?

**Comment by lilyminium** [comment]
Yeah other ways to "select" atoms from empty AtomGroups also return an empty AtomGroup, so having `select_atoms` do the same seems more consistent.
```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF
>>> u = mda.Universe(PSF)
>>> u.atoms[[]][[]]
<AtomGroup with 0 atoms>
```

>  I can’t remember, is there a way to go to universe
again via global or something?

I think `group` is an AtomGroup so you could try `group.universe`, if that's what you're asking. Anyway it looks like it's just the StringSelection with this issue and the below seems to fix it, although it may be superceded by #2755?

```diff
--- mask |= [fnmatch.fnmatch(x, val) for x in values]
+++ mask |= np.asarray([fnmatch.fnmatch(x, val) for x in values], 
+++                    dtype=np.bool)
```

**Comment by orbeckst** [comment]
Is this a *fix* that should be backported #2798 or is it a *Change* that can only go into 2.0.0?

**Comment by orbeckst** [comment]
Looking at PR #2767 you seem to be considering it a fix so I'll add it to the backport #2798. This will also add a CHANGELOG entry.
